src/Altinn.App.Core/Infrastructure/Clients/Notifications/Sms/SmsNotificationClient.cs

```diff
@@ -0,0 +1,97 @@
+using System.Net.Http.Headers;
+using System.Text.Json;
+using Altinn.App.Core.Configuration;
+using Altinn.App.Core.Internal.App;
+using Altinn.App.Core.Internal.Notifications.Sms;
+using Altinn.App.Core.Models.Notifications.Sms;
+using Altinn.Common.AccessTokenClient.Services;
+using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.Core.Infrastructure.Clients.Notifications.Sms;
+
+internal sealed class SmsNotificationClient : ISmsNotificationClient
+{
+    private readonly ILogger<SmsNotificationClient> _logger;
+    private readonly IHttpClientFactory _httpClientFactory;
```

Comment:
I saw that it was used once, so that's why I dared do it myself too 😄 
My rationale: object lifecycle and allocation is easier to reason about - I know `SmsNotificationClient` and `IHttpClientFactory` are singletons, whenever I create and dipose the clients the caching mechanism for underlying handlers in `IHttpClientFactory` have the chance to do their thing. With typed clients, the lifetime is an implementation detail (transient atm), so the lifetime of the client (and when the factory can manage the lifetime of the handlers) depends on the surrouding scope (e.g. if the client was actived from a scoped or root `IServiceProvider`)

I usually try to make code do as little work as possible aside from performing the desired function (i.e. simple code). Transient lifetime for instance is not required and generates noise (in the form of memory allocation and code that runs at times I don't really care for 😄).